quipucords/api/credential/view.py

```diff
@@ -110,39 +37,120 @@ class Meta:
-            cred.delete()
-            return Response(status=status.HTTP_204_NO_CONTENT)
-
-        except Credential.DoesNotExist as exception:
-            raise Http404 from exception
+    serializer_by_type = {
+        DataSources.NETWORK: SshCredentialSerializerV2,
+        DataSources.OPENSHIFT: AuthTokenOrUserPassSerializerV2,
+        DataSources.VCENTER: UsernamePasswordSerializerV2,
+        DataSources.SATELLITE: UsernamePasswordSerializerV2,
+        DataSources.ANSIBLE: UsernamePasswordSerializerV2,
+        DataSources.RHACS: AuthTokenSerializerV2,
+    }
+
+    def get_input_serializer_class(self, cred_type):
+        """Get the type-specific serializer for processing create/update inputs."""
+        if serializer := self.serializer_by_type.get(cred_type):
+            return serializer
+        raise ValueError("Invalid credential type.")
+
+    def create(self, request, *args, **kwargs) -> Response:
+        """
+        Create a credential.
+
+        That this method is based on rest_framework.mixins.CreateModelMixin.create.
+        Modifications were made only to use different input and output serializers.
+        """
+        cred_type = request.data.get("cred_type")
+        input_serializer_class = self.get_input_serializer_class(cred_type)
+
+        input_serializer = input_serializer_class(data=request.data)
+        input_serializer.is_valid(raise_exception=True)
+        instance = input_serializer.save()  # Save the validated data.
+        # We are not using `self.perform_create(input_serializer)` here because we need
+        # the instance to pass through a different serializer for output.
+
+        # Use the common output serializer for the response.
+        output_serializer = self.get_serializer_class()(instance)
+        headers = self.get_success_headers(output_serializer.data)
+        return Response(
+            output_serializer.data, status=status.HTTP_201_CREATED, headers=headers
+        )
+
+    def update(self, request, *args, **kwargs) -> Response:
+        """
+        Update a credential.
+
+        That this method is based on rest_framework.mixins.UpdateModelMixin.update.
+        Modifications were made only to use different input and output serializers.
+        """
+        partial = kwargs.pop("partial", False)
+        instance = self.get_object()
+        cred_type = request.data.get("cred_type", instance.cred_type)
+        input_serializer_class = self.get_input_serializer_class(cred_type)
+        input_serializer = input_serializer_class(
+            instance, data=request.data, partial=partial
+        )
+        input_serializer.is_valid(raise_exception=True)
+        self.perform_update(input_serializer)
+
+        if getattr(instance, "_prefetched_objects_cache", None):
+            # If 'prefetch_related' has been applied to a queryset, we need to
+            # forcibly invalidate the prefetch cache on the instance.
+            instance._prefetched_objects_cache = {}
```

Comment:
well thought out! I wasn't considering the prefetched objects could be an issue for patch/put